test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,633 @@
+            logger.info(f"OUTPUT: Message creation response status: {message_response.status_code}")
+            assert message_response.status_code == 200, "Failed to create test message."
+
+        # Retrieve chats with a request for 20 chats
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,
+            "page": 1,
+            "page_size": 20  # Requesting more chats than exist
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Check for successful response
+        assert response.status_code == 200, "Failed to retrieve chats."
+
+        response_data = response.json()
+        messages = response_data.get('messages')
+        assert isinstance(messages, list), "Expected 'messages' to be a list."
+
+        # Check for the total number of user messages, excluding the initial question
+        user_messages = [msg for msg in messages if
+                         msg['sender_type'] == 'customer' and msg['message'] not in ["Initial question",
+                                                                                     "You will get the correct answer once AI is integrated."]]
+
+        assert len(user_messages) == len(messages_to_add), (
+            f"Expected {len(messages_to_add)} user messages, but got {len(user_messages)}. "
+            f"Retrieved messages: {user_messages}"
+        )
+
+        # Verify that all added messages are in the response
+        for expected_message in messages_to_add:
+            assert any(msg.get('message') == expected_message for msg in user_messages), (
+                f"Expected message '{expected_message}' not found in the response."
+            )
+
+        logger.info("=== Test Case 13: Retrieve 20 chats when only 5 exist Completed ===\n")
+
+    def test_retrieval_with_specific_pagination(self):
+        """Test case: Add 5 chats, retrieve 2 chats on the first page, 2 on the second, and 1 on the third."""
+        logger.info("=== Test Case 14: Specific Pagination Retrieval ===")
+
+        # Step 2: Create a new chat to ensure valid chat_id
+        create_chat_url = f"{self.BASE_URL}/chat"
+        initial_question = "initial_question"
+        create_chat_data = {
+            "customer_guid": self.valid_customer_guid,
+            "question": initial_question
+        }
+        response = requests.post(create_chat_url, json=create_chat_data)
+        logger.info(f"Chat creation response status: {response.status_code}, {response.text}")
+        self.assertEqual(response.status_code, 200, "Failed to create a new chat.")
+
+        # Get the new chat_id
+        self.valid_chat_id = response.json().get("chat_id")
```

Comment:
Use a local chat_id variable if you are creating one in the function

Reply:
I also used the local chat ID for this test case, sir.